src/Akka.Streams.Kafka/Stages/Consumers/Actors/KafkaConsumerActor.cs

```diff
@@ -572,11 +586,57 @@ private void ReceivePoll(Internal.Poll<K, V> poll)
             }
         }
 
+        private void CommitAggregatedOffsets()
+        {
+            if(_commitMaps.Count == 0) return;
+            var aggregatedOffsets = AggregateOffsets(_commitMaps);
+            // commits can occur after the partition has been revoked from the consumer, so ensure that we only attempt to
+            // commit partitions that are currently assigned to the consumer. For high volume topics, this can lead to small
+            // amounts of replayed data during a rebalance, but for low volume topics we can ensure that consumers never appear
+            // 'stuck' because of out-of-order commits from slow consumers.
+            var assignedOffsetsToCommit = aggregatedOffsets.Where(kvp => _consumer.Assignment.Contains(kvp.Key))
+                .Select(c => new TopicPartitionOffset(c.Key, c.Value)).ToImmutableSet();
+            var replyTo = _commitSenders;
+            // flush the data before calling `consumer.commit`
+            _commitMaps = ImmutableList<TopicPartitionOffset>.Empty;
+            _commitSenders = ImmutableHashSet<IActorRef>.Empty;
+            Commit(assignedOffsetsToCommit, replyTo);
+        }
+
+        public static IReadOnlyDictionary<TopicPartition,Offset> AggregateOffsets(IReadOnlyCollection<TopicPartitionOffset> offsets)
+        {
+            var aggregate = new Dictionary<TopicPartition, Offset>();
+            foreach(var offset in offsets)
+            {
+                if (aggregate.TryGetValue(offset.TopicPartition, out var existingOffset))
+                {
+                    if (existingOffset < offset.Offset)
+                        aggregate[offset.TopicPartition] = offset.Offset;
+                }
+                else
+                {
+                    aggregate.Add(offset.TopicPartition, offset.Offset);
+                }
+            }
+
+            return aggregate;
+        }
+        
+        private void CommitAndPoll()
+        {
+            var refreshOffsets = _commitRefreshing.RefreshOffsets;
+            if (refreshOffsets.Any())
+            {
+                _log.Debug("Refreshing committed offsets: {0}", refreshOffsets.JoinToString(", "));
+                Commit(refreshOffsets, ImmutableHashSet<IActorRef>.Empty);
+            }
+            Poll();
+        }
+
         private void Poll()
         {
             var currentAssignment = _consumer.Assignment.ToImmutableList();
-            var initialRebalanceInProcess = _rebalanceInProgress;
-
+            CommitAggregatedOffsets();
```

Comment:
Aggregated offsets get committed next as this will help drive some of the polling behavior (i.e. what messages are available beyond our currently committed offsets for each partition?)